### PR TITLE
deps: bump to Nuxt 3.12, bring other dependencies up-to-date, separate prettier in repo & checks, update devcontainer.json format and docker container to Node.js to 20-bookworm

### PR DESCRIPTION
**Pull request by HappyNTH**
PR predominately updates dependencies that had steadily grown out-of-date and resolves the various breaking issues that these version updates encounter. The biggest ticket is perhaps Nuxt 3.2 -> 3.12, but there are a host of other changes that can be read directly from the [package.json file diff](https://github.com/BristolSTA/uobtheatre-web/pull/419/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519).

In addition, [Prettier](https://prettier.io/) has been removed from `eslint`, split out into its own `yarn` command, and integrated into the GitHub CI flow, as [recommended by Prettier](https://prettier.io/docs/en/integrating-with-linters).

`Dockerfile` Node.js specification has been bumped from `18-buster` to `20-bookworm`, as some dependencies were starting to drop support for v18.

Further, `devcontainer.json` is updated to use the new `customizations/vscode/extensions` format for VSCode extentions. The `Volar` TypeScript Vue plugin is removed, as it is depreciated and wrapped into the main `Vue` plugin.

### CODE REVIEW

File: tests/unit/support/helpers/mount.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { mount as vtuMount } from '@vue/test-utils';
-import { Router, _RouteLocationBase } from 'vue-router';
+import type { Router, _RouteLocationBase } from 'vue-router';
 import { vi } from 'vitest';
+//@ts-ignore TODO: Remove this ignore when @nuxtjs/apollo is updated to support bundler
```
Comment (zst-c) [inline tests/unit/support/helpers/mount.ts:4]: Is this linked to an issue somewhere so we remember to do as much? Or can we do so?

**Review by zst-c — APPROVED** [review]
Looks great! As you say, not actually too massive - just a lot of formatting, `type`s being expressed explicitly, and removing some trailing test commas.

All good ((:

File: classes/Ticket.ts [hunk 2]
```diff
@@ -56,7 +56,9 @@ export default class {
 
   static dataFromQRCode(rawQRCode: string) {
     try {
-      const result = JSON.parse(atob(rawQRCode));
+      const result = JSON.parse(
+        Buffer.from(rawQRCode, 'base64').toString('ascii')
```
[the remark this answers is not in the document]
Comment (HappyNTH) [reply classes/Ticket.ts:60]: Good spot! Resolved

File: plugins/sentry.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { SentryLink } from 'apollo-link-sentry';
 
 export default defineNuxtPlugin((nuxtApp) => {
   const runtimeConfig = useRuntimeConfig();
-  const environment = nuxtApp.$config.ENV;
+  const environment = nuxtApp.$config.ENV as string;
```
[the remark this answers is not in the document]
Comment (HappyNTH) [reply plugins/sentry.ts:7]: I believe this resolves the Sentry environment key stored on the server's `.env`, for reporting errors as they occur

**Pull request merged**